lib/luacrypto_shash.c

```diff
@@ -0,0 +1,298 @@
+/*
+* SPDX-FileCopyrightText: (c) 2025 jperon <cataclop@hotmail.com>
+* SPDX-License-Identifier: MIT OR GPL-2.0-only
+*/
+
+/***
+* Low-level Lua interface to the Linux Kernel Crypto API for synchronous
+* message digest (hash) algorithms, including HMAC.
+*
+* This module provides a `new` function to create SHASH transform objects,
+* which can then be used for various hashing operations.
+*
+* @module crypto_shash
+*/
+
+#define pr_fmt(fmt) KBUILD_MODNAME ": " fmt
+
+#include <linux/kernel.h>
+#include <linux/module.h>
+#include <linux/crypto.h>
+#include <crypto/hash.h>
+#include <linux/err.h>
+#include <linux/slab.h>
+
+#include <lua.h>
+#include <lualib.h>
+#include <lauxlib.h>
+#include <lunatik.h>
+
+static const lunatik_class_t luacrypto_shash_tfm_class;
+
+typedef struct {
+	struct shash_desc *sdesc; /* Points to kmalloc'ed shash_desc + space */
+} luacrypto_shash_tfm_t;
+
+LUNATIK_PRIVATECHECKER(luacrypto_check_shash_tfm, luacrypto_shash_tfm_t *);
+static void luacrypto_shash_tfm_release(void *private)
+{
+	struct crypto_shash *tfm;
+	luacrypto_shash_tfm_t *tfm_ud = (luacrypto_shash_tfm_t *)private;
+
+	if (tfm_ud->sdesc) {
```

Comment:
I would put this in a local variable.. I think it would clean up a bit the code..